### PR TITLE
2.6.0 Release Candidate

### LINKED ISSUE #45: MySQL container needs explicit configuration
**Describe the bug**
The MySQL container doesn't perform any configuration which can lead to some issues. 

1. The max_packet_size is not set which limits the users ability to import larger databases
2. Memory limits and related settings are not configured for innodb_buffer_pool which can lead to poor performance for larger databases

**Expected behavior**
The MySQL container should be a bit closer to what a typical server deployment might look like to ensure it is stable and useful to the end user.

**Environment information**
 - All environments are equally impacted but macOS and Windows will need to meet "minimum requirements" to ensure enough memory is allocated to Docker Engine.

**Additional context**
I will be providing a PR to allow for configuration of MySQL and set Docker memory limits

### PR DESCRIPTION
**Pull request by christianc1**
### Description of the Change

- All new containers based on CentOS 7 for PHP 5.6 through 7.1 and CentOS 8 for PHP 7.2 through 7.4
- Adds support for PHP 7.4.
- Enable/Disable Xdebug using ENV var in the docker-compose.yml file. Disabling Xdebug improves performance significantly on some systems.
- Additional tools installed including git, composer, telnet, nc and more.
- opcache configured with optimal settings for development.
- php-fpm now runs as www-data on macOS and Windows hosts or as the user who created the project on Linux. This is to ensure permissions are correct.
- Upgrade to Elasticsearch 5.6.16.
- Limit MySQL to 1GB of memory, configure for 1GB memory limit.
- Limit Elasticsearch to 1GB of memory, configure for 1GB of memory.
- Improved image update handling.
- Brings in #60 resolving shell escaping issues
- Brings in #63 resolving issues pulling the phpmemcachedadmin utility

### Benefits

All new php-fpm container that is smaller and significantly improves performance when Xdebug is disabled. A number of bug fixes and quality of life improvements.

### Possible Drawbacks

The MySQL and Elasticsearch containers will enforce a 1GB memory limit. Some larger sites may suffer and we're actively looking for feedback in this area.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Applicable Issues

Closes #45, #52, #60 , #61 and #63. Should improve speed issues brought up in #25 (would enjoy feedback to confirm)

### CODE REVIEW

File: src/environment.js [hunk 1]
```diff
@@ -229,6 +229,96 @@ const upgradeEnv = async function( env ) {
...
+	// Upgrade image.
+	let phpVersion = yaml.services.phpfpm.image.split(':').pop();
+
+	if ( '5.5' === phpVersion ) {
+		console.warn( 'Support for PHP v5.5 was removed in the latest version of WP Local Docker.' );
+		console.error( 'This environment cannot be upgraded.  No changes were made.' );
+
+		process.exit(1);
+	}
+	upgraded.services.phpfpm.image = `10up/wp-php-fpm-dev:${phpVersion}`;
```
Comment (dustinrue) [inline src/environment.js]: please lookup the php container like in create.js line 263:

```
'image': images[`php${answers.phpVersion}`],
```

Comment (dustinrue) [reply src/environment.js]: Also note there is a special case for a Linux system. See create.js line 290

Comment (christianc1) [reply src/environment.js]: We don't have access to the `answers` in the upgrade command.  `10updocker upgrade` does not prompt.

Comment (christianc1) [reply src/environment.js]: oh, i see what you mean, I'll adjust

Comment (christianc1) [reply src/environment.js]: fixed in 48dbb84

**Comment by christianc1** [comment]
Updated docs in https://github.com/10up/wp-local-docker-docs/pull/16

**Pull request merged**